### PR TITLE
Nofruit are completely random and have a static cycling animation.

### PR DESCRIPTION
**Pull request by N3X15**
# Executive Summary
This gets rid of the reflex-based selection of fruits in No-Fruit and No-Fruit Pies, makes the cycling animation a static DMI-based animation (based off of a random selection of 20 icons), and gets rid of the clicks altogether.

# Justification
* The code-based cycling animation has lead to lag and even client crashes in some cases when enough nofruit are on-screen.
* The reflex-based selection of harvest produce has been demonstrated to be gamable via autoclickers and memorization of cycles.  In addition, the presence of exotic fruits in the produce list is worrisome at best.  By making the selection completely random and independent from the icon on-screen, the output is truly random.
* The fucking clicking makes me want to stab things.

# Icons
Icons were made by a BYONDTools script that selects 20 icons at random from food.dmi and harvest.dmi.

# Changelog
:cl:
* tweak: Produce from nofruit is no completely random and has no bearing from the icon on screen.
* sounddel: Nofruit no longer clicks.
* bugfix: Nofruit cycling animations no longer produce lag.

### CODE REVIEW

**Comment by kilozombie** [comment]
I've written several billion words on my thoughts on this, so you can refer to earlier PRs if you want additional discussion.

This is a good fix for the _lag_, but awful for gameplay, in my mind. Fix the lag in some other way. No-fruit pie doesn't need this nerf.

This does not fix gaming no-fruit. This only 'fixes' gaming no-fruit pie, which again, does not need this nerf.

This does not solve no-fruit powergaming. You can still grow a thousand no-fruit and get whatever you want. Congratulations-- zero people were actually aiming for no-fruit, and now zero people are affected. Instead, anyone who enjoyed no-fruit pie gambling is shafted-- 100% random is not as fun as having some skill to it.

edit: Why are you describing it as a reflex-based selection? You _literally_ cannot react in time to get what you want because it ticks too quickly. You _have_ to memorize it currently. Merge https://github.com/d3athrow/vgstation13/pull/13699 instead, it adds content, forces skill, and was made by the person who actually created no-fruit in the first place.

**Comment by N3X15** [comment]
For one, this nerfs both the pie and the harvestable.

Second, I don't give two flabby shits over who made what.  This is an open-source project, and people can and will stick their chubby little dickbeaters into whatever code that want to fix.  

Third, this gets rid of both reflex, memory, AND skill, making it RNG-based.  I really don't know why the fuck people want to make this particular plant able to be gamed, but it isn't right.  It should be purely random, like the slot machines.

**Comment by kilozombie** [comment]
>For one, this nerfs both the pie and the harvestable.

When did I say otherwise? I understand that. That's why I oppose this PR, for one. The pie does NOT need a nerf. I cannot comprehend why you, or anybody else, thinks it needs a nerf.

>Second, I don't give two flabby shits over who made what

Considering shadowmech said he intended for no-fruit to be, in some way, skill-based, it's valuable to listen to his input on it. He knows more about no-fruit and its intended purpose than you do. The ease is bad, being able to get what you want easily is bad, but when it requires hitting an item within an 0.4s window without being able to expect it... that's not easy, that's hard. That's a good thing.

>Third, this gets rid of both reflex, memory, AND skill, making it RNG-based. 

Why?

>It should be purely random, like the slot machines.

Why?

Justify these things for me. Why is it harmful that a guy can have a no-fruit pie (that takes some effort to get), and with a very high degree of practice and skill, maybe have a chance at getting a differently-shaped cake to eat? Who is harmed by this?

This change is illogical, and also, frankly, rude to anyone who uses these features.

>I really don't know why the fuck people want to make this particular plant able to be gamed, but it isn't right.

Hey, N3X15, refer to fig. 1-6 on this PR to see how easy it is to game even when you attempt, futilely, to stop it. https://github.com/d3athrow/vgstation13/pull/13697

Why do coders try and edit things and features they don't know anything about?

**Comment by N3X15** [comment]
Well, here's the problem,  Even if we cast all the gamability shit out of the argument purely because "that's how they were INTENDED", the fucking thing is a huge performance issue.  Being in a room with only 5 of the things going can be a massive FPS drop and introduces networking lag, as well.  How would you propose to fix that?  Reducing the cycles per second will help, but only marginally increases the number of them on-screen before you lag out.

**Comment by kilozombie** [comment]
The speed nerf will bump it from 5 to 20 max, assuming we're going by that metric. Make no-fruit auto-whack themselves if they're on a tile with another no-fruit and keep the ticking sound.
Not a perfect solution. It also doesn't remove a feature.

**Comment by Exxion** [comment]
1. Make the static animation cycle in a specific order
2. Save the time when it starts
3. Subtract it from the time at which it gets hit
4. Pick appropriate fruit

**Comment by despotate** [comment]
[I](https://github.com/d3athrow/vgstation13/pull/13606)

[want](https://github.com/d3athrow/vgstation13/pull/13613)

[to](https://github.com/d3athrow/vgstation13/pull/13649)

[get](https://github.com/d3athrow/vgstation13/issues/13712)

[off](https://github.com/d3athrow/vgstation13/pull/13697)

[the no-fruit](https://github.com/d3athrow/vgstation13/pull/13699)

[wild ride](https://github.com/d3athrow/vgstation13/pull/13786)

**Comment by kilozombie** [comment]
>he submitted the first PR

This is the future you chose

**Comment by CptWad** [comment]
I like this PR because it fixes all the problems and keeps the fruit "random"
also this game needs less things that you can powergame. Trying to "game" the system for your benifit is not fun. 
My only gripe is r  💀  p clicking

**Comment by kilozombie** [comment]
>Trying to "game" the system for your benifit is not fun.

Uh oh, that guy down in Botany is making no-fruit pie so he can make chocolate cake more easily. It takes a long time and it's unreliable! (Nobody cares.)

Even shadowmech said this wasn't gaming the system if you're using timing and skill instead of, say, putting your mouse on some particular pixel.

>also this game needs less things that you can powergame

Sure. I can agree with that. **Too bad this PR does not fix no-fruit powergaming**

**Comment by CptWad** [comment]
this isnt a super duper game breaking balance pr. This is a **rare** item that allows you to get other, **somewhat rarer** items in a **random** fashion. To me this PR aims to upkeep that. Yeah you cant time yourself to get the exact item you wished for, but so what? Why is that such a big deal. Its not like your job as a chef is ruined just because you cant make free cakes from no fruit pies. Hell it would be more fun if you just gave out the pies and let other players get random things from a carrot cake to a brain cake. I could also argue the whole muh rp element, that even if an item exists that could randomly turn into other items and defy the laws of physics, under what circumstances would trying to "game" that improve the scenario?
I dont quite understand why you are so worked up over this

>This gets rid of the reflex-based selection of fruits in No-Fruit and No-Fruit Pies, makes the cycling animation a static DMI-based animation (based off of a random selection of 20 icons)

sounds like a fix to me

**Comment by kilozombie** [comment]
>Hell it would be more fun if you just gave out the pies and let other players get random things from a carrot cake to a brain cake.

Fun gimmick. But I have an even more fun one: challenge people to land on a specific item, and give them a prize if they land on it. I've done this as Trader and it's marvelous.

Or, as a more general thing, get _really_ good at no-fruit pie whacking, and get 3-4 of the same rare food item for a gimmick. Now, let's note here for a second that this would take a load of time, since with shadowmech's change it would be very difficult to get specific items, OR a lot of skill. In addition, there is almost no rare food in this game that actually has a use outside of Being Eaten For Fun.

What this PR is is it's Feature Removal. Shadowmech's change adds a new feature: no-fruit pie aim games.

>sounds like a fix to me

It changes something that didn't need a change, and fails to fix something that might need a fix. This PR fixes the lag, nothing else. It does so by sacrificing something I personally like a lot, and a lot of people agree with me there.

>I dont quite understand why you are so worked up over this

Feature removal is shitty. It's shittier when the guy who's pressing forth the PR doesn't even know what the fuck he's talking about. At least despotate had a stake in it and knew what he was arguing about; you and N3X15 need to understand the problems which are actually here, and what solutions have already been put forth for them.

Killing fun for arbitrary, bad reasons, filled with circular logic "no-fruit should be random because it should be random because that's how it is" makes me very, very mad.

I'm going to begin posting this at the end of every message: **This PR does not fix no-fruit powergaming.**

**Comment by CptWad** [comment]
Yeah sure fine you win that argument.
However I still support this pr based on that it improves performance and reduces soundspam
also reviewing past prs you said that this method was fun and simple so long as it fixed the lag even if it did remove the timed clicks here:
image

**Comment by kilozombie** [comment]
I was just afraid that no-fruit would be completely gimped in that PR. As more discussion came up, and I realized more and more merits of having it be possible to time it or aim for it, I decided to argue the point that I really wanted (that no-fruit pie really doesn't need a huge nerf like this).

(This PR does not fix no-fruit powergaming, nor has any PR in the past which named itself 'removes no-fruit powergaming')

**Comment by Kammerjunk** [comment]
Never compromise
Not even in the face of armageddon

**Comment by kilozombie** [comment]
We had a chat in OOC.

If you legitimately want to fix no-fruit, this isn't the way to do it.

If you insist on some kind of change, which _remains my least favorite thing to do,_ try this:

- Make no-fruit unable to get trader plants, xenoarch plants. This fixes no-fruit powergaming for the most part because it's just as feasible to mutate your plants if you want mutant plants, and more accurate.
- Either keep it as it originally was, or give it an 0.4s delay and make it cycle through lists. This is so no-fruit retains that fun aiming factor.
- To prevent lag, no-fruit whacks itself if it's picked while near other no-fruit, due to some kind of interference or some shit

Maybe there are problems with this solution, but I don't like the idea of fixing a bug by removing a feature, while trying to argue that the feature was bad in the first place, and then changing your mind because "well I'm fixing a bug". Stop movin' those goalposts.

The lag is a bug. Fix it without messing things up. _Please._

**Comment by N3X15** [comment]
The problem is that in order to fix the lag, we have to fix the way the no-fruit was implemented.  With the way BYOND works, having it be aimable will produce lag no matter what.  All you're doing by increasing the delay is making the server able to handle slightly more of them.  The self-whacking is extremely hacky and weird behaviour, and counter-intuitive.  Many others in OOC agreed that having it be random would be a good compromise.

**Comment by kilozombie** [comment]
>Many others in OOC agreed that having it be random would be a good compromise.

People who learned about the issue 5 minutes ago and were still uninformed, jesus christ. After you left and I explained the situation, everyone previously making an impulse "yes" vote shifted towards my take on things.

You are being unreasonably stubborn. Emojicracy has actually spoken and a wide majority of people with any investment (that play the game, that play Botany, that play Trader) would not like this change, I don't think, considering all the discussion that's gone on. Work out a solution, I beg. I am *begging*, because I am sick and fucking tired of this "balance" meme. There is a solution that will please people, and I would really like if you spent more time looking for it.

Self-whacking is hacky and weird, but not counter-intuitive, not any more than no-fruit itself. And it would _absolutely_, definitely fix the lag problem if done correctly, without removing aimable.

**Comment by Exxion** [comment]
I just said how to do it without lag

**Comment by N3X15** [comment]
I did explain it with you.  Patiently.  Probe also explained it to you.  You're correct, I did leave to implement the suggestions that were made and may have missed out on some discussion.  Instead, I come back to find that you still insist, after being told multiple times why this isn't a good change, that it's your way or the highway.

So, I'm going to paste this every time you say something until you come up with new arguments:

1. No-fruit are a tremendous source of lag.  This is because their design dictates that they sent at least 2 packets to the client: One for the image change, the other for the click sound.
2. To make them able to be aimed, updates must be sent to the client so the client knows the current state of the no-fruit.  This is the reason behind the design portion of point 1.
3. Even if we made a static DMI animation of the sequence, lag would result in desynchronization of the state of the server-side no-fruit and the client-side DMI animation.
4. Aimability of the no-fruit being the intent of its creation is not an argument, because the technical limitations of BYOND can and do override design considerations. 
5. Players have been known and have admitted to using this system to gain access to fruits they desire without difficulty, including fruits not obtainable normally. Do you want botany growing only no-fruit, or would you rather they try things the normal way?
6. Delaying the updates sent by the no-fruit would only delay the onset of serious lag issues.
7. "Self-whacking" is destructive to the fruit and counter-intuitive, not to mention a being hacky workaround of a serious design flaw.

**Comment by kilozombie** [comment]
>Players have been known and have admitted to using this system to gain access to fruits they desire without difficulty, including fruits not obtainable normally. Do you want botany growing only no-fruit, or would you rather they try things the normal way?

Proof that you are blind to all discussion or reason. Here's proof, right here in the pudding. You already said not 15 minutes ago in OOC that the only problem with no-fruit is lag, and here you are trying to argue this point again. I even made you repeat it.

Here I am, having said that there can be something to prevent that. 

Here you are, making the change I recommended that removes xenoarch plants and vox plants from the list, therefore making this point moot already.

Thanks.

>"Self-whacking" is destructive to the fruit and counter-intuitive, not to mention a being hacky workaround of a serious design flaw.

For real? What the fuck are you saying here? Destructive to the fruit? How is it more counter-intuitive than any other part of no-fruit? How is it _destructive to the fruit_? What does that mean?

It is a _fix_ to a _problem_ that doesn't _remove shit._ I never said it was perfect, I'm arguing that I'd rather have a slightly hacky (but explainable through lore) fix than one that removes something I really like.

**Comment by Exxion** [comment]
The clicking is kind of dumb

One of the time vars (too tired to remember which one) is lag-independent
Unless you mean clientside lag would cause the desync

**Comment by N3X15** [comment]
Looks like I forgot a few of the same old talking points you've been hammering for the past two days.

1. No-fruit are a tremendous source of lag.  This is because their design dictates that they sent at least 2 packets to the client: One for the image change, the other for the click sound.
2. To make them able to be aimed, updates must be sent to the client so the client knows the current state of the no-fruit.  This is the reason behind the design portion of point 1.
3. Even if we made a static DMI animation of the sequence, lag would result in desynchronization of the state of the server-side no-fruit and the client-side DMI animation.
4. Aimability of the no-fruit being the intent of its creation is not an argument, because the technical limitations of BYOND can and do override design considerations. 
5. Players have been known and have admitted to using this system to gain access to fruits they desire without difficulty, including fruits not obtainable normally. Do you want botany growing only no-fruit, or would you rather they try things the normal way?
6. Delaying the updates sent by the no-fruit would only delay the onset of serious lag issues.
7. "Self-whacking" is destructive to the fruit (meaning the fruit is destroyed when it self-whacks) and counter-intuitive, not to mention a being hacky workaround of a serious design flaw.
8. This would solve powergaming by making it entirely chance-based rather than memory- or skill-based.

Yes, @Exxion, any lag would result in the timing becoming increasingly "off" from what the players' client shows.  DMI animations are entirely client-side and have no bearing on the server at all, so after the animation starts, there's no guarantee that the state of the no-fruit will match the state of the sprite on the client.  The current no-fruit ensures this by forcing the client to display a single static non-animated image.  The next tick displays the next state's image.

**Comment by kilozombie** [comment]
>This would solve powergaming by making it entirely chance-based rather than memory- or skill-based.

I'm going to step back a moment so I don't flip my shit, just a sec

**Comment by kilozombie** [comment]
I've been fighting these PRs for 10 days now. I'm getting to be a little miserable, because miscommunication is permeating with every word I try and say. Consistently, the only thing that has become clear to me is that no PR, no PR out of all of them, has fixed no-fruit powergaming completely.

Here is why, comprehensively.

- You can still grow enough no-fruit to get whatever you want.
- Growing this much no-fruit is not very time-intensive.
- Even if it is random what you get from no-fruit, you can easily get all of it with a small quantity of plants.
- In five minutes, or ten minutes, you can have plenty of varieties of plants.
- If xenoarch and vox seeds are included, then plenty of those are going to be obtained during this.
- This will stay consistent no matter how random you make the cycle.

Now, I don't think that this is powergaming, in most senses of the word. You're still putting effort in. You're putting effort into a different thing and an alternative method, but it's still effort, and still quite a bit of it.

But the fact that you, Probe, despotate, _everyone_ is claiming to be to fixing that... is ridiculous. Because you aren't.

You aren't. Flat out.

**Comment by N3X15** [comment]
* Okay, and if you have that much no-fruit in the first place, you're probably already late-game.
* See 1.
* No really, judging from the size of the list generated.  There's a few hundred possibilities.
* Yeah, and?
* They're not included, see commit added today.
* Again, so?

My point is that by aiming, you have a far greater chance of getting what you want, because you know what you have to do.  With the random system, you have to do far more work in order to get what you want, and it's not *guaranteed*.

**Comment by kilozombie** [comment]
>Okay, and if you have that much no-fruit in the first place, you're probably already late-game.

Wrong. Gotten hundreds of no-fruit within 30 minutes of roundstart.

>No really, judging from the size of the list generated. There's a few hundred possibilities.

We're talking about no-fruit, not no-fruit pie. Are you illiterate?

Wait.

WAIT.

Wait.

Let's step back a second.

You think that there are _hundreds_ of varieties of plants.

Ha.

_Hahahaha._

**Comment by N3X15** [comment]
Sounds like I should reduce the yield of no-fruit, then.

**Comment by kilozombie** [comment]
**_The yield of no-fruit is 1._**

This is why people get mad at coders. Not because they're salty, but because the people coding _feature removal_ do not take even _SECONDS_ of time to research what they're removing or what any of it means. 

They do not even look at the code to see _any_ part of what they're nerfing and realize that they are on a completely different page than everybody that plays the game. 
They do not understand anything about a department, yet make changes to it, and while still misunderstanding the basic, core principles of the argument, try to inject their own shitty circular arguments to make themselves sound right.

This whole time, you have not even known the difference about no-fruit and no-fruit pie, and yet you're trying to mess with both.

Unfathomable. Inconceivable.

I have no idea why people were excited to have you back, but I hope you fuck right off again.

**Comment by N3X15** [comment]
And instead of helping us understand more, you prefer to sit there and screech about how unfair and cruel it is that people who spend more time fixing shit than playing the game don't know some things.

**Comment by kilozombie** [comment]
>don't know some things

How about, you don't have a clue what we are even arguing about?

There is nothing wrong with not knowing shit. I don't know a lot of shit. But I don't try and involve myself in an argument I don't know anything about. I don't try and change other people's departments that I don't know anything about.

**Comment by N3X15** [comment]
And yet here you are, assuming to know everything there is to know about game design and every botanist's personal feelings on how they want plants to work.

**Comment by kilozombie** [comment]
There are enough people who have come and spoken or voted on this problem to understand that there's a general majority on it, for people that know what's being talked about.

I don't know everything about game design-- and am sorry if I come across that way. You know much more than me on that front. Doesn't mean this isn't an issue worth talking about from a perspective of somebody who plays the job.

**Comment by N3X15** [comment]
Just because there's a majority doesn't mean the majority understands the problem and the choices available.

**Comment by kilozombie** [comment]
For people that know what's being talked about, again, there's been a majority.

Even the emojicracy is mostly filled (on both sides) with people very familiar with the issue, who have posted in previous threads or seen them all.

Of course a server poll would catch a lot of people uninformed. But on github, with people that have seen my screeching for days now, seems to me most people are informed or have stopped tuning in to vote.

**Comment by clusterfack** [comment]
This isn't an emojiocracy

**Comment by kilozombie** [comment]
Sorry, I forgot. The only metric for whether something should be merged or not is legitimate argument, and I've put mine forth. If coders decide against the emojicracy and against my legitimate arguments and criticism, as well as other people's legitimate arguments and criticism, then I suppose there's no real sanity to how things get merged at all besides "well I don't agree and I'm a coder, so... *merge*"

**Comment by N3X15** [comment]
Yes, which is why I'm sitting here futilely trying to argue against someone who won't listen to reason.  I have the power to self-merge this, and yet, for some reason, I don't feel like having you barking at me for the next ten years.

The fact of the matter is that this is how they should have been made from the start.  Just making them aimable requires they be made with the .icon_state spamming, *which is the entire fucking problem*.

**Comment by kilozombie** [comment]
Not if there's only one going at a time. Unnoticable lag. Trying to introduce another would self-whack it.

Look, man. If you self-merge, I'm not going to bark at you. I'm just going to accept that it's pretty much hopeless to try here. I _understand_ what you're saying. I understand it causes lag. I'm offering a solution that fixes the lag without removing the feature, because I don't want the feature removed, and I also think removing the lag would be a cool thing to do.

**Comment by N3X15** [comment]
That's the problem, though, there are almost NEVER just one of these things sitting there.  With this change, you can HAVE 50 of them in botany without crashing.

**Comment by kilozombie** [comment]
With the proposed self-whack change, it makes it impossible to have more than one ticking, as that's an annoyance and loud and laggy.

**Comment by Shadowmech88** [comment]
Wouldn't making them immutable, and therefore unable be altered to produce tons of fruit in no time at all, *also* heavily reduce lag by preventing botanists from flooding botany with them, while still keeping the skill-based-reward trait of the fruit?

**Comment by kilozombie** [comment]
Certainly. It would take _ages_ to get a laggy number of fruits.

You would get a much greater effect by just mutating something else and mass-producing it anyway. You can stop clients with 3000-4000 items of _any_ type on one tile, and 2,000 is feasible within 45 minutes or so IIRC.

So making it immutable is yet another solution.

**Comment by N3X15** [comment]
I can incorportate immutability to this PR, certainly.  That alone still wouldn't solve the flawed design, as someone with enough time and planters could still produce enough of them to cause lag.

**Comment by kilozombie** [comment]
Let's calm down with the number of nerfs we implement for no-fruit. Botanists are already killing themselves hard enough over the PR as it is.
Implement self-whacking and maybe immutability if you'd like a solution to the lag. Include that with shadowmech's change so you can still aim for things.
Easy.

**Comment by Shadowmech88** [comment]
Someone with enough time could produce lag in any number of ways.
I'm trying to offer solutions that might be used *instead* of reducing the no-fruit to a slot machine. Of course allowing players to choose the result is always going to be able to *theoretically* cause terrible lag, but I'm sure most players would be fine with toning it down to the point where it rarely *actually* causes terrible lag.

**Comment by Shadowmech88** [comment]
At the very least, I think we should *try* such solutions before taking such drastic measures as this PR.

**Comment by kilozombie** [comment]
>Someone with enough time could produce lag in any number of ways.

Very good point. I think the key is not potential, but how likely it's gonna happen, how easy it is. Good way to remove how easy it is might simply be immutability, as that's the real key for getting large quantities.

**Comment by PJB3005** [comment]
Ok.

I have the solution to No-Fruit.

Remove it from the game entirely.

Can't be abused when it's not in game!

**Comment by CptWad** [comment]
I am a botanist and I'm not killing myself over this

**Comment by despotate** [comment]
[The only _real_ solution to all potential forms of powergaming, including the one that no other PR solves (massing no-fruit with xenobotany and clipping) is to make them a random and **truly limited** item](https://github.com/d3athrow/vgstation13/pull/13649). There are NO plants in botany that are truly limited at the moment so it would be nice if we could use no-fruit as a proof-of-concept for it; right now, if you have one seed of any plant, within five minutes you can have at least a hundred seeds.

Re-implementation of the above closed PR can, of course, be left to someone else because while immutability and clipping are easy fixes (setting the flag and a low enough health), preventing the grown from being turned into multiple seeds through the extractor isn't.

File: code/modules/reagents/reagent_containers/food/snacks/grown.dm [hunk 1]
```diff
@@ -19,6 +21,12 @@
 		src.pixel_x = rand(-5, 5) * PIXEL_MULTIPLIER
 		src.pixel_y = rand(-5, 5) * PIXEL_MULTIPLIER
 
+/proc/get_special_fruits(var/filter=HYDRO_PREHISTORIC|HYDRO_VOX)
+	. = list()
+	for(var/type in existing_typesof(/obj/item/weapon/reagent_containers/food/snacks/grown))
```
Comment (PJB3005) [inline code/modules/reagents/reagent_containers/food/snacks/grown.dm]: You can cast this variable to `/obj/item/weapon/reagent_containers/food/snacks/grown` and then you'll be able to remove the `:` below.

File: code/modules/reagents/reagent_containers/food/snacks/grown.dm [hunk 2]
```diff
@@ -19,6 +21,12 @@
...
+		if((initial(type:hydroflags) & filter)>0)
```
Comment (PJB3005) [inline code/modules/reagents/reagent_containers/food/snacks/grown.dm]: Wouldn't just `if (initial(type:hydroflags) & filter)` work too?

**Review by PJB3005 — DISMISSED** [review]

**Comment by kilozombie** [comment]
And preventing all forms of _potential_ powergaming just because it's feasibly possible is dumb to me. Potential shouldn't guide feature removal. You can misuse a lot of things in this game, doesn't mean it should all be gimped.

**Comment by N3X15** [comment]
@PJB3005 Fixed.

**Comment by N3X15** [comment]
This isn't about solving all sources of powergaming.  This is about fixing a major performance problem with a poorly designed game element, which also happens to make powergaming that particular game element more difficult.  Your PR merely puts a band-aid on the problem.

I really don't know why you want to die on this particular hill, but you seemed absolutely determined to turn this into a gigantic spectacle for no reason whatsoever.

**Comment by kilozombie** [comment]
I dislike things I like being removed. Is that not a good enough reason?

Just looking for a compromise here. I'm past the point of being stubborn, I'm going a ways to try and get a fix that makes both of us at least kinda happy.

**Comment by Shadowmech88** [comment]
I've implemented the sound removal and exclusion list from this PR into #13699, in the interest of finding the best solution that everyone can agree on.

**Comment by 9600bauds** [comment]
Why the fuck was this self-merged?

File: code/modules/reagents/reagent_containers/food/snacks/grown.dm [hunk 3]
```diff
@@ -19,6 +21,12 @@
 		src.pixel_x = rand(-5, 5) * PIXEL_MULTIPLIER
 		src.pixel_y = rand(-5, 5) * PIXEL_MULTIPLIER
 
+/proc/get_special_fruits(var/filter=HYDRO_PREHISTORIC|HYDRO_VOX)
+	. = list()
+	for(var/T in existing_typesof(/obj/item/weapon/reagent_containers/food/snacks/grown))
+		if(initial(T:hydroflags) & filter)
```
Comment (9600bauds) [inline code/modules/reagents/reagent_containers/food/snacks/grown.dm:27]: :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

**Comment by Durgen** [comment]
@9600bauds 
kilo and nexis has a spat in ooc, nexis threatened to just self merge it, kilo dared him to do it, nexis did

**Comment by kilozombie** [comment]
I was explaining no-fruit to somebody and N3X15 interpreted it as being towards him. I said "hey man i would like to be civil since i've been bad at that lately", he told me off, and said if I said another word he'd self-merge. I pointed out that self-merging is bad and would get an insta-revert, he did it anyway.

**Comment by PJB3005** [comment]
Ok.

Proper solution.

OOC logs.

Now.

**Comment by kilozombie** [comment]
Sure.

[20:24:33]OOC: Ivar Robotnik/N3X15 (@260,232,1): Because the guy who coded it is a moron
[20:25:01]OOC: Kilozombie/Kilozombie (@8,81,2): OOC: N3X15: Because the guy who coded it is a moron
[20:25:12]OOC: Kilozombie/Kilozombie (@8,81,2): shadowmech is a moron for putting certain plants in the list...?
[20:25:35]OOC: Kilozombie/Kilozombie (@8,81,2): i don't want to start aguments anymore but that's kinda rude to the dude
[20:25:47]OOC: Ivar Robotnik/N3X15 (@276,242,1): He didn't put some plants on the list
[20:25:54]OOC: Ivar Robotnik/N3X15 (@276,242,1): He put ALL PLANTS on the list
[20:26:19]OOC: Kilozombie/Kilozombie (@8,81,2): i mean, that didn't seem like a bad idea at the time
[20:26:34]OOC: Kilozombie/Kilozombie (@8,81,2): only came up later as a problem to people
**(somebody says people abused it by aiming for specific plants, i call him an idiot as a joke and explain that that's not how it happened)**
[20:27:05]OOC: Kilozombie/Kilozombie (@8,81,2): anyway the real way people got all the plants was just growing a lot of no-fruit
[20:27:16]OOC: Kilozombie/Kilozombie (@8,81,2): with shadowmech's PR, you can't grow a lot of no-fruit and you can't get xenoarch or trader plants
**(N3X15 thinks this was directed towards him)**
[20:27:51]OOC: Ivar Robotnik/N3X15 (@384,240,2): No, I'm done arguing about this.  It's been four fucking days and you won't stop yapping.
[20:28:37]OOC: Kilozombie/Kilozombie (@8,81,2): i'm not trying to antagonize you at all n3x or start an argument, just saying you don't need to call shadowmech a moron
[20:28:50]OOC: Kilozombie/Kilozombie (@8,81,2): maybe a little civility would go a ways, considering how shitty i've been at that
[20:29:37]OOC: Ivar Robotnik/N3X15 (@384,240,2): Don't care.  Shut the fuck up.  You've been yelling about a stupid small part of botany for four goddamn days and I'm done with you.
[20:29:59]OOC: Kilozombie/Kilozombie (@8,81,2): thanks, alrighty
[20:30:05]OOC: Kilozombie/Kilozombie (@8,81,2): love ya too
[20:30:15]OOC: Ivar Robotnik/N3X15 (@381,239,2): A single goddamn word more and I merge my PR.
[20:30:32]OOC: Kilozombie/Kilozombie (@8,81,2): do it, do the one thing that will guarantee an insta-revert
[20:30:40]OOC: Ivar Robotnik/N3X15 (@381,237,2): Done.
[20:30:48]OOC: Kilozombie/Kilozombie (@8,81,2): jesus christ

**Comment by 9600bauds** [comment]
c my comment on https://github.com/d3athrow/vgstation13/pull/13699#issuecomment-277549311

**Pull request merged**